GiEnJul/ClientApp/src/components/admin/AdminPage.tsx

```diff
@@ -1,23 +1,89 @@
-import { withAuthenticationRequired } from "@auth0/auth0-react";
+import { useAuth0, withAuthenticationRequired } from "@auth0/auth0-react";
 import { Container } from "@material-ui/core";
 import * as React from "react";
+import { useEffect, useState } from "react";
 import LoadingPage from "../../common/components/LoadingPage";
+import InstitutionMacro from "../institution/InstitutionMacro";
 import LogOutButton from "../login/LogOutButton";
 import AdminMenu from "./common/AdminMenu";
 import Completed from "./connections/Completed";
 import Suggested from "./connections/Suggested";
 
+type UserDataType = {
+  app_metadata?: {role:string},
+  email?: string,
+  email_verified?: boolean,
+  identities?: [{}],
+  last_ip?: number,
+  last_login?: string,
+  logins_count?: number,
+  name?: string,
+  nickname?: string,
+  picture?: string,
+  updated_at?: string,
+  user_metadata?: {string:string}
+}
+
+
 function AdminPage() {
-  return (
-    <>
-      <LogOutButton></LogOutButton>
-      <Container maxWidth="xl">
-        <AdminMenu />
-        <Suggested />
-        <Completed />
-      </Container>
-    </>
-  );
+  const { user, getAccessTokenSilently } = useAuth0();
+
+  const [userData, setUserData] = useState<UserDataType | undefined>(undefined);
+  const [isLoaded, setIsLoaded] = useState<boolean>(false);
+
+
+
+  async function getUserInformation(): Promise<UserDataType> {
+    
+    const domain = process.env.REACT_APP_DEV_TENANT_AUTH0!;
+
+        const accessToken = await getAccessTokenSilently({
+          audience: `https://${domain}/api/v2/`,
+          scope: "read:current_user",
+        });
+
+        const userDetailsByIdUrl = `https://${domain}/api/v2/users/${
+          user!.sub
+        }`;
+
+        const  response  = await fetch(userDetailsByIdUrl, {
+          headers: {
+            Authorization: `Bearer ${accessToken}`,
+          },
+        });        
+        const userInformation: UserDataType = await response.json();
+        return userInformation;
+
+  }
+  useEffect(() => {
+      getUserInformation().then((response: UserDataType) => setUserData(response)),
+      setIsLoaded(true)
+  },[]);
+
+
+
+  if (!isLoaded) {
+    return <LoadingPage />;
+  } else if (isLoaded && userData == null) {
```

Comment:
Could be placed with initial "if" as they return same value

Reply:
Not really. That is if you are Loaded and the code couldn't fetch the userData. So it should display an errorpage, but we don't have an errorpage yet. So the return is just a temp

Reply:
I see 👍 then its probably a good idea to throw an exception or handle the error in a temporary way, instead of leaving a "functional" endpoint where it should fail.

Reply:
Fixed with d931b9c